main/level_1.gd

```diff
@@ -42,11 +44,19 @@ func _on_enemy_spawner_timer_timeout() -> void:
 		random_y += enemy_buffer_spawn_positive
 
 	new_enemy.position = Vector2(random_x, random_y)
-	add_child(new_enemy)
+	enemy_container.add_child(new_enemy)
 	
 	new_enemy.call_deferred("initialize_enemy", player)
 
 func _on_player_shoot(Bullet: PackedScene, _position: Vector2, _direction: Vector2) -> void:
 	var new_bullet = Bullet.instantiate()
 	add_child(new_bullet)
 	new_bullet.call_deferred("start", _position, _direction)
+
+func _on_waves_timer_timeout() -> void:
+	for child in enemy_container.get_children():
+		if child.has_method("queue_free"):
```

Comment:
I don't think you need this verification, I think every node has that, but sure :)